ogr/ogrsf_frmts/miramon/ogrmiramondriver.cpp

```diff
@@ -0,0 +1,249 @@
+    {
+        if (!poDS->Open(poOpenInfo->pszFilename, nullptr, nullptr,
+                        poOpenInfo->eAccess == GA_Update,
+                        poOpenInfo->papszOpenOptions))
+        {
+            delete poDS;
+            poDS = nullptr;
+        }
+    }
+
+    return poDS;
+}
+
+/****************************************************************************/
+/*                         OGRMiraMonDriverCreate()                              */
+/****************************************************************************/
+
+static GDALDataset *
+OGRMiraMonDriverCreate(const char *pszName, CPL_UNUSED int /*nBands*/,
+                       CPL_UNUSED int /*nXSize*/, CPL_UNUSED int /*nYSize*/,
+                       CPL_UNUSED GDALDataType /*eDT*/, char **papszOptions)
+{
+    OGRMiraMonDataSource *poDS = new OGRMiraMonDataSource();
+
+    if (poDS->Create(pszName, papszOptions))
+        return poDS;
+
+    delete poDS;
+    return nullptr;
+}
+
+/****************************************************************************/
+/*                           RegisterOGRMM()                                */
+/****************************************************************************/
+
+void RegisterOGRMiraMon()
+
+{
+    if (GDALGetDriverByName("MiraMonVector") != nullptr)
+        return;
+
+    GDALDriver *poDriver = new GDALDriver();
+    poDriver->SetDescription("MiraMonVector");
+    poDriver->SetMetadataItem(GDAL_DCAP_VECTOR, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_CREATE_LAYER, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_CREATE_FIELD, "YES");
+    poDriver->SetMetadataItem(GDAL_DMD_LONGNAME,
+                              "MiraMon Vectors (.pol, .arc, .pnt)");
+    poDriver->SetMetadataItem(GDAL_DMD_EXTENSION, "pol");
+    poDriver->SetMetadataItem(GDAL_DMD_EXTENSIONS, "pol arc pnt");
+    poDriver->SetMetadataItem(GDAL_DMD_HELPTOPIC,
+                              "drivers/vector/miramon.html");
+    poDriver->SetMetadataItem(GDAL_DCAP_VIRTUALIO, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_Z_GEOMETRIES, "YES");
+
+    poDriver->SetMetadataItem(
+        GDAL_DMD_OPENOPTIONLIST,
+        "<OpenOptionList>"
+        "  <Option name='Height' scope='vector' type='string' "
+        "   description='Sets which of the possible heights is chosen: "
+        "   the first, the highest or the lowest one.'>"
+        "    <Value>First</Value>"
+        "    <Value>Lower</Value>"
+        "    <Value>Highest</Value>"
+        "  </Option>"
+        "  <Option name='iMultiRecord' scope='vector' type='string' "
+        "   description='Sets which of the possible records is chosen: "
+        "   0, 1, 2,... or the Last one. Use JSON when a serialized "
+        "   JSON is wanted'>"
+        "    <Value>0</Value>"
+        "    <Value>1,...</Value>"
+        "    <Value>Last</Value>"
+        "    <Value>JSON</Value>"
+        "  </Option>"
+        "  <Option name='OpenMemoryRatio' scope='vector' type='float' "
```

Comment:
I'm slightly concerned by making this available to end users. This looks scary. Does that really make a performance difference when you play with that? Isn't the default good enough?

Reply:
For the OpenMemoryRatio I'm not so sure thant the CreateMemoryRatio.
Let me make some analysis and I'll tell you

Reply:
Ok, you win :) I test it and I was wrong. So I deleted this and the creation option one.